### PR TITLE
[Enhancement] Optimize avg(distinct column) with CTE

### LINKED ISSUE #7262: Optimize avg(distinct column) with CTE
## Enhancement
In #5919， StarRocks support count(distinct column) and sum(distinct column) with CTE, we can also support avg(distinct column) with CTE

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7262 

## Problem Summary(Required) ：

```
Mark count(distinct v1) as cd(v1), avg(distinct v1) as ad(v1), sum(distinct v1) as sd(v1).
Before:
          Agg[ad(v1), sum(v4), max(v5)]
                       |
                   Child Plan
 
  After:
                     CTEAnchor
                    /           \
            CTEProduce          project[ad(v1) = sd(v1) / cd(v1)]
                 /                  \
            Child Plan           Cross join
                                /          \
                        Cross join       Agg[sum(v4), max(v5)]
                         /          \            \
                  Agg[cd(v1)]     Agg[sd(v1)]   CTEConsume
                      /               \
                 CTEConsume        CTEConsume
```

### CODE REVIEW

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithoutGroupByRule.java [hunk 1]
```diff
@@ -114,11 +144,77 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
                     OptExpression.create(new LogicalJoinOperator(JoinOperator.CROSS_JOIN, null), left, right);
             allCteConsumes.offerFirst(join);
         }
+        Operator.Builder builder = OperatorBuilderFactory.build(allCteConsumes.get(0).getOp());
+        Operator newOperator = builder.withOperator(allCteConsumes.get(0).getOp()).
+                setProjection(new Projection(columnRefMap)).build();
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithoutGroupByRule.java:149]: use project node

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 51 / 51 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithoutGroupByRule.java|51|51|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 50 / 50 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithoutGroupByRule.java|50|50|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**